pkg/server/commands/listusers/list_users_rpc_test.go

```diff
@@ -3281,3 +3637,91 @@ func TestListUsersConfig_MaxConcurrency(t *testing.T) {
 		})
 	}
 }
+
+func TestInvertRelationshipStatus(t *testing.T) {
+	require.Equal(t, HasRelationship, NoRelationship.InvertRelationshipStatus())
+	require.Equal(t, NoRelationship, HasRelationship.InvertRelationshipStatus())
+}
+
+func TestListUsers_ExpandExclusionHandler(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+
+	ds := memory.New()
+	t.Cleanup(ds.Close)
+
+	t.Run("avoid_producing_explicitly_negated_subjects", func(t *testing.T) {
```

Comment:
This fails right now and demonstrates we're producing more subjects than we should from the exclusion handler. For example, we should only produce `user:*` with the tuples and model provided, but we end up producing 3-4 subjects (`user:*` and multiple copies of `user:jon`).

<img width="1145" alt="Screenshot 2024-05-20 at 9 14 58 AM" src="https://github.com/openfga/openfga/assets/2899204/5913a033-7ca6-4c15-b55f-2f434d577360">